### DISCUSSION TITLE
Why am I getting the wrong id for the current panel?

### DISCUSSION THREAD

**Discussion by shebaoting** [filamentphp/filament#14195] [Help]
### Package

Panel builder

### Package Version

3.0

### How can we help you?

http://scrm.test/dev
This panel is dev, but the id I get is the id of another panel, customer.
```
$currentPanel = Filament::getCurrentPanel();
$currentPanelId = $currentPanel->getId();
  dd($currentPanelId);
```

<img width="433" alt="image" src="https://github.com/user-attachments/assets/c153f5ce-f8b6-4525-8132-1fd6749325aa">

**Comment by leandrocfe** [comment]
I don’t think you can get the current panel at that point. It might make more sense to use custom middleware in your panels.

```php
$panel
...
->middleware([ CustomMiddleware::class ])
...
```

CustomMiddleware.php
```php
public function handle(Request $request, Closure $next): Response
{
    $currentPanel = filament()->getCurrentPanel();
    $currentPanel->getId();

    return $next($request);
}
```

**Comment by e2d2-dev** [comment]
depends when/where are using it. a closure could do the job.

**Comment by shebaoting** [comment]
I need to determine this in the AppServiceProvider. But in AppServiceProvider, filament()->getCurrentPanel()->getId() is not able to get the exact panel id

**Reply by e2d2-dev** [reply]
you just need to just a closure for that

**Reply by shebaoting** [reply]
> you just need to just a closure for that

Can you give me a rough example? Thank you very much.

**Reply by e2d2-dev** [reply]
`public function handle(Request $request, Closure $next): Response
{
    $currentPanel = **fn() =>** filament()->getCurrentPanel();
    $currentPanel->getId();

    return $next($request);
}`
